### PR TITLE
Use Tempfile for sockets and state files in tests

### LINKED ISSUE #2150: Open all state/sock files in their own temporary directory, and rm them after receiving stop/kill signal
I was just wondering why my tests were failing locally, then I realized that I had the sock/state files in my test directory because they hadn't been cleaned up properly:

```
/test/test_puma.state
/test/test_server.sock
/test/test_control.sock
```

I used to like not having these in .gitignore because I use `git status` to see if my current state is "clean", but now I have to look for those socket files manually (which is kinda hard because when alphabetically sorted in the test dir they really blend in).

Thoughts anyone?

### PR DESCRIPTION
**Pull request by alexeevit**
### Description
Closes #2150 

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` the pull request title.
- [x] I have added appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: test/helpers/integration.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 require "puma/control_cli"
 require "open3"
+#require_relative "../helper"
```
Comment (dentarg) [inline test/helpers/integration.rb:5]: Don't think we need this line? :)

**Comment by dentarg** [comment]
> Closes #2150

I don't think "... and rm them after receiving stop/kill signal" that #2150 begs for, is handled here? (I'm not sure what a good way to handle that looks like). Nonetheless, this looks like a great start!

**Comment by alexeevit** [comment]
> > Closes #2150
> 
> I don't think "... and rm them after receiving stop/kill signal" that #2150 begs for, is handled here? (I'm not sure what a good way to handle that looks like). Nonetheless, this looks like a great start!

Well, I've done one possible implementation of the management of created files. Not sure if it's good design to redefine `#run` but I let you judge 😄

**Comment by dentarg** [comment]
Hehe, creative! :)

I'll let @nateberkopec be the judge 😛

**Comment by nateberkopec** [comment]
Yeah, I don't think `run` is the right place to inject here. I think there are probably other places in Minitest you could take a look at, probably in the signal handling code.

**Comment by alexeevit** [comment]
> Yeah, I don't think `run` is the right place to inject here. I think there are probably other places in Minitest you could take a look at, probably in the signal handling code.

You're right, I found a better place to put my hooks

File: test/helpers/tmp_path.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+module TmpPath
+  def capture_exceptions
+    super
+  rescue
+    clean_tmp_paths
+    raise
+  ensure
+    clean_tmp_paths
```
Comment (nateberkopec) [inline test/helpers/tmp_path.rb:8]: ensure is always evaluated, so you don't need the clean_tmp_paths in `rescue`.

**Review by nateberkopec — CHANGES_REQUESTED** [review]
quick fix, otherwise I think it looks brilliant now.

File: test/helpers/tmp_path.rb (hunk 2 above)
Comment (alexeevit) [reply test/helpers/tmp_path.rb:5]: Fixed

**Pull request merged**